streams/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala

```diff
@@ -572,4 +595,44 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
         list.reverse must_=== (1 to 4).toList
       }
     )
+
+  import scalaz.zio.duration._
+
+  private def repeat =
+    unsafeRun(
+      Stream(1)
+        .repeat(Schedule.recurs(4))
+        .run(Sink.collect[Int])
+        .map(_ must_=== List(1, 1, 1, 1, 1))
+    )
+
+  private def repeatShortCircuits =
+    unsafeRun(
+      for {
+        ref <- Ref.make[List[Int]](Nil)
+        _ <- Stream
+              .fromEffect(ref.update(1 :: _))
+              .repeat(Schedule.spaced(10.millis))
+              .take(2)
+              .run(Sink.drain)
+        result <- ref.get
+      } yield result must_=== List(1, 1)
+    )
+
+  private def spaced =
+    unsafeRun(
+      Stream(1, 2, 3)
+        .spaced(Schedule.recurs(1))
+        .run(Sink.collect[Int])
```

Comment:
I'm only starting to see how awesome this API is 😍

Reply:
Going to be even nicer with the upcoming restructuring around `ZManaged` :-)